### PR TITLE
Refactor/cost estimate and product item

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Purpose of this PR**
This PR refactors the `CostEstimate` Class to the same builder pattern as the `Offer` Class. 
Additionally it adds the previously missing fields `ExpirationDate` and `ProjectManager` to the `CostEstimate `class 
and removes the ` computeTotalCost `method  from the `ProductItem` class making `ProductItem` a proper dto.
Finally it introduces simple tests for the introduced and changed classes. 

**Misc**
This PR also deprecates the `Person` class to mark it for future removal, since i was unsure if that tag in itself would be worth a seperate PR.

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy:39]: In which order do you plan to merge the PRs of yours. The one introducing the projectmanager object should be either way reflected here at some point. In this PR or in the other one so that we won't miss it :)

File: src/main/groovy/life/qbic/datamodel/people/Person.groovy
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/people/Person.groovy:3]: Did we want to deprecate this Person class? I think we don't even use it in the context of TOMATO.

File: src/test/groovy/life/qbic/datamodel/dtos/business/ProductItemSpec.groovy
Comment (jenniferboedker) [inline src/test/groovy/life/qbic/datamodel/dtos/business/ProductItemSpec.groovy:29]: What happens when I open the box? 👀

**Review by jenniferboedker — CHANGES_REQUESTED** [review]
I added some comments especially because of the choice of the Person class to be deprecated, is it the correct one? Furthermore, I think we should try to structure all of your PRS since we need to include the ProjectManager class also into the new CostEstimate.

After we decided on how to approach on how to merge your PRs I will approve.

Other than that, nice work! LGTM

File: src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy:39]: Good Question, I was unsure how to proceed here since the projectManager Class is in a seperate PR and i wanted to make sure this PR works properly in it's current state. 
I'd rather merge the other pr into development and merge that back into this PR and adapt the property to make sure everything runs properly.

File: src/main/groovy/life/qbic/datamodel/people/Person.groovy
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/people/Person.groovy:3]: I think it would be a good idea to deprecate classes with the same name but different implemenations to avoid confusion in the future. I don't know in which other classes this class is used, therefore we discussed deprecating it and removing it as soon as we know how easy it is to do so

File: src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy
Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy:39]: Thats a good idea, then I will wait for your merge until my final approval :)

File: src/test/groovy/life/qbic/datamodel/dtos/business/ProductItemSpec.groovy
Comment (Steffengreiner) [reply src/test/groovy/life/qbic/datamodel/dtos/business/ProductItemSpec.groovy:29]: Then all the IDEs in the world will stop working and we'll have to code in the terminal directly 😨

File: src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/datamodel/accounting/CostEstimate.groovy:39]: 👍

File: src/test/groovy/life/qbic/datamodel/dtos/business/CustomerSpec.groovy
Comment (jenniferboedker) [inline src/test/groovy/life/qbic/datamodel/dtos/business/CustomerSpec.groovy:42]: Since this is the customer spec I would keep the customer here.. if you dont want to keep the customer here I would suggest to update the comment :)

Comment (jenniferboedker) [reply src/test/groovy/life/qbic/datamodel/dtos/business/CustomerSpec.groovy:42]: I would keep the old test, as I see you created a ProjectManagerSpec so no need to include the project manager to the customer spec

**Review by jenniferboedker — COMMENTED** [review]
Left some minor comment about your change for one test, if you address this, this PR is approved 👍  good work

File: src/test/groovy/life/qbic/datamodel/dtos/business/CustomerSpec.groovy
Comment (Steffengreiner) [reply src/test/groovy/life/qbic/datamodel/dtos/business/CustomerSpec.groovy:42]: No one saw that... >< Nice Catch!

**Review by jenniferboedker — APPROVED** [review]
LGTM

**Pull request merged**
